server/src/main/java/org/apache/iotdb/db/metadata/template/Template.java

```diff
@@ -412,90 +385,81 @@ public boolean isPathExistInTemplate(String path) throws IllegalPathException {
   }
 
   public boolean isDirectNodeInTemplate(String nodeName) {
-    if (templateRoot.hasChild(nodeName)) {
-      return true;
-    } else {
-      return false;
-    }
+    return directNodes.containsKey(nodeName);
   }
 
   public boolean isPathMeasurement(String path) throws IllegalPathException {
     String[] pathNodes = MetaUtils.splitPathToDetachedPath(path);
-    IMNode cur = templateRoot;
-    for (String node : pathNodes) {
-      if (cur.hasChild(node)) {
-        cur = cur.getChild(node);
-      } else return false;
+    if (!directNodes.containsKey(pathNodes[0])) {
+      throw new IllegalPathException(path, "Path does not exist.");
     }
```

Comment:
There's PathNotExistException. May use this is better.

Reply:
thanks! all this kind of exception now changed to PathNotExistException